cpp-client/src/aiaautils.cpp

```diff
@@ -0,0 +1,439 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *  * Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ *  * Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *  * Neither the name of NVIDIA CORPORATION nor the names of its
+ *    contributors may be used to endorse or promote products derived
+ *    from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS ``AS IS'' AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+ * OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "../include/nvidia/aiaa/log.h"
+#include "../include/nvidia/aiaa/exception.h"
+
+#include <itkResampleImageFilter.h>
+#include <itkConnectedComponentImageFilter.h>
+#include <itkLabelShapeKeepNObjectsImageFilter.h>
+#include <itkConstantPadImageFilter.h>
+#include <itkRegionOfInterestImageFilter.h>
+#include <itkLinearInterpolateImageFunction.h>
+#include <itkNearestNeighborInterpolateImageFunction.h>
+
+#include <itkImage.h>
+#include <itkImageFileReader.h>
+#include <itkImageFileWriter.h>
+#include <itkImageIOFactory.h>
+
+#include <algorithm>
+#include <sstream>
+#include "../include/nvidia/aiaa/aiaautils.h"
+
+namespace nvidia {
+namespace aiaa {
+
+template<class TImage>
+typename TImage::Pointer resizeImage(typename TImage::Pointer image, typename TImage::SizeType targetSize, bool linearInterpolate) {
+
+  auto imageSize = image->GetLargestPossibleRegion().GetSize();
+  auto imageSpacing = image->GetSpacing();
+
+  using ImageType = TImage;
```

Comment:
ImageType for clarity of following codes
and TImage is for shorter function signature?

Reply:
yes..